### PR TITLE
deploy: remove leading and trailing spaces from images

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/14659

### CODE REVIEW

**Comment by mfojtik** [comment]
@tnozicka PTAL

File: pkg/deploy/util/util.go [hunk 1]
```diff
@@ -266,6 +266,13 @@ func MakeDeployment(config *deployapi.DeploymentConfig, codec runtime.Codec) (*a
 		return nil, fmt.Errorf("couldn't clone podSpec: %v", err)
 	}
 
+	// Fix trailing and leading whitespace in the image field
+	// This is needed to sanitize old deployment configs where spaces were permitted but
+	// kuberneted 3.7 tightened the validation of container image fields.
```
Comment (tnozicka) [inline pkg/deploy/util/util.go]: s/kuberneted/kubernetes
maybe a ref to a PR making this change

Comment (mfojtik) [inline pkg/deploy/util/util.go]: fixed.

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
re-[test]

**Comment by mfojtik** [comment]
[merge][severity:blocker]

**Pull request merged**